openmc/geometry.py

```diff
@@ -455,3 +455,105 @@ def get_lattices_by_name(self, name, case_sensitive=False, matching=False):
         lattices = list(lattices)
         lattices.sort(key=lambda x: x.id)
         return lattices
+
+    def count_cell_instances(self, universe=None):
```

Comment:
Sure, I like that design, it's just as we implemented this algorithm in its original form in OpenMOC. Though I think `count_cell_instances()` should be a private `Universe` class method since we don't want users randomly calling it on any universe they wish.